extensions/bitwarden/src/api.ts

```diff
@@ -0,0 +1,38 @@
+import execa from "execa";
+
+export class BitwardenApi {
+  private env: Record<string, string>;
+  constructor(clientId: string, clientSecret: string) {
+    this.env = {
+      BW_CLIENTSECRET: clientSecret,
+      BW_CLIENTID: clientId,
+    };
+  }
+
+  async syncItems(sessionToken: string): Promise<void> {
+      await this.exec(["sync", "--session", sessionToken])
+  }
+
+  async login(): Promise<void> {
+     await this.exec(["login", "--apikey"]);
+  }
+
+  async listItems<ItemType>(type: string, sessionToken: string): Promise<ItemType[]> {
+    const {stdout} = await this.exec(["list", type, "--session", sessionToken])
+    return JSON.parse(stdout)
+  }
+
+  async unlockVault(password: string): Promise<string> {
+      const {stdout: sessionToken} = await this.exec(["unlock", password, "--raw"])
+      return sessionToken
+  }
+
+  async getVaultStatus(sessionToken: string | undefined): Promise<"unlocked" | "locked" | "unauthenticated"> {
+    const {stdout} = await this.exec(sessionToken ? ["status", "--session", sessionToken]: ["status"]);
```

Comment:
```suggestion
    const {stdout} = await execa("bw", ["status", "--session", sessionToken]);
```
This should be enough. I'd rather we simplify calls to the `bw` command as much as possible, and keep error handling within the SDK logic.

Reply:
The ternary is not literally error handling, but with my suggested changes the `stdout` variable may contain an error. This error should be handled within the `status` function.